main.cpp

```diff
@@ -91,6 +91,116 @@ good to go!
  Wait for my code review.
  */
 
+ #include <iostream>
+
+struct FloatType
+{
+    float add(float lhs, float rhs);
+    float subtract(float lhs, float rhs);
+    float multiply(float lhs, float rhs);
+    float divide(float lhs, float rhs);
+};
+
+float FloatType::add(float lhs, float rhs)
+{
+    return lhs + rhs;
+}
+
+float FloatType::subtract(float lhs, float rhs)
+{
+    return lhs - rhs;
+}
+
+float FloatType::multiply(float lhs, float rhs)
+{
+    return lhs * rhs;
+}
+
+float FloatType::divide(float lhs, float rhs)
+{
+    if (lhs == 0.0f || rhs == 0.0f)
```

Comment:
it doesn't matter if `lhs` is 0.  it only matters if `rhs` is 0.  if `rhs` is 0, then you'll actually have a division by zero.
- [ ] revise